ODLC state unit test

Ooglely opened the pull request:
A unit test for the ODLC state so that we can test its functionality as a standalone state.

Commit:
in progress

micjfet committed:
update file comparisons

micjfet committed:
file copy fix

micjfet committed:
first pre-commit fixes

micjfet committed:
precommit and other errors fixed

micjfet committed:
precommit and other errors fixed

micjfet committed:
some run errors fixed

micjfet committed:
some run fixes (not yet working)

micjfet committed:
multiprocessing fishes

Flamingpuffins committed:
small docstring fix for find_odlcs function

Flamingpuffins committed:
Merge branch 'feature/odlc_unit_test' of https://github.com/MissouriMRR/SUAS-2023 into feature/odlc_unit_test

Flamingpuffins committed:
Merge branch 'develop' into feature/odlc_unit_test

Flamingpuffins committed:
Added sim flag check to the find_oldcs function

Flamingpuffins committed:
fixed sim flag check in find_odlc

Flamingpuffins committed:
Added normal move to if simflag is true

Commit:
some multiprocessing logging and fixes

micjfet committed:
removed test statements

Ooglely committed:
Fix async errors, remove print statements

Ooglely commented:
Waiting on the ability to skip states in the state machine before ready for review

Ooglely committed:
Add waypoint state skip

Ooglely committed:
Merge branch 'develop' into feature/odlc_unit_test

Flamingpuffins commented on flight/test_files/odlc_state_test.py:
```diff
@@ -0,0 +1,57 @@
+    """
+    asyncio.run(state_machine.run())
+
+
```
add docstring

Flamingpuffins committed:
Added odlc unit test docstrings

cjhr95 commented on state_machine/states/impl/waypoint_impl.py:
```diff
@@ -102,7 +105,8 @@ async def run(self: Waypoint) -> State:
             curr_altitude: float = drone_position.relative_altitude_m
-            altitude_slope: float = (waypoint.altitude - curr_altitude) / path_length
+            # altitude_slope: float = (waypoint.altitude - curr_altitude) / path_length
+            # removing this because "too many local variables" eww
```
Change or remove this comment for professionalism's sake.

cjhr95 commented on flight/test_files/odlc_state_test.py:
```diff
@@ -0,0 +1,68 @@
+    state_machine_process.start()
+
+    activated_odlcs = 0
+    while activated_odlcs != 5:
```
Change this to a variable or a constant; hard-coded values are usually a no-go

Ooglely committed:
Resolve comments

Ooglely committed:
Merge branch 'develop' into feature/odlc_unit_test

Ooglely committed:
Merge branch 'develop' into feature/odlc_unit_test

Commit:
mypy fixes for state_tracker and odlc_impl

Commit:
mypy fix for flight_manager.py

micjfet committed:
Merge branch 'develop' into feature/odlc_unit_test

cjhr95 commented on flight/test_files/odlc_state_test.py:
```diff
@@ -0,0 +1,69 @@
+        Whether or not the test is being run in a simulation
+    """
+    logging.basicConfig(level=logging.INFO)
+    drone = Drone()
+    flight_settings = FlightSettings(sim_flag=_sim, skip_waypoint=True)
+    state_machine = StateMachine(Start(drone, flight_settings), drone, flight_settings)
```
type hints here

cjhr95 commented on flight/test_files/odlc_state_test.py:
```diff
@@ -0,0 +1,69 @@
+    )
+    state_machine_process.start()
+
+    activated_odlcs = 0
```
Needs type hint

cjhr95 commented on flight/test_files/odlc_state_test.py:
```diff
@@ -0,0 +1,69 @@
+    while activated_odlcs != odlc_count:
+        try:
+            with open("flight/data/output.json", "r", encoding="UTF-8") as file:
+                output_data = json.load(file)
```
Need a type hint here as well

cjhr95 reviewed (changes requested):
Missing a few, quick type hints

Commit:
type hints in odlc_state_test.py

Commit:
"changes from develop"
Merge branch 'feature/odlc_unit_test' of https://github.com/MissouriMRR/SUAS-2023 into feature/odlc_unit_test

cjhr95 reviewed (approved):
Nice job kiddos

Ooglely committed:
Merge branch 'develop' into feature/odlc_unit_test
